lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1828,7 +2153,31 @@ def main():
             docker_api_version='1.25',
             detect_usage=_detect_healthcheck_start_period,
             usage_msg='set healthcheck.start_period'
-        )
+        ),
+        update_config_max_failure_ratio=dict(
+            docker_py_version='2.1.0',
+            docker_api_version='1.25',
+            detect_usage=lambda c: c.module.params['update_config']['max_failure_ratio'] is not None,
```

Comment:
This will most likely die if `update_config` is not specified. You probably have to do something like
```suggestion
            detect_usage=lambda c: (c.module.params['update_config'] or {}).get('max_failure_ratio') is not None,
```

Reply:
(Note that `detect_usage` will only be called when the conditions for `docker_py_version` or `docker_api_version` are not satisfied. That's why you won't notice this until you downgrade to an earlier docker-py version or an earlier docker daemon. I've already patched the existing checks here: https://github.com/ansible/ansible/pull/53129/commits/ed914fc4da4f4d6b2ab7f450536063f4a334225d; that's part of #53129)